`Feature2D.detectAndCompute` `mask` argument is missing `| None` in type stubs

### System Information

- OpenCV Python version: 4.12.0.88 
- MacOS: 15.5
- Python version: 3.13.1

### Detailed description

In the Python bindings, the 2D Feature detectors that implement `detectAndCompute` (`AKAZE`, `AffineFeature`, `BRISK`, `KAZE`, `ORB` and `SIFT`) support `mask=None` ([OpenCV Tutorial](https://docs.opencv.org/4.x/da/df5/tutorial_py_sift_intro.html)), but the type stubs do not reflect this.

Based on cbf3b1187d9192d64511c2af01fb2aa2310b069d, 0e17a879d77a0d79caf1c05d238fc17d3eee4ddc, and fc810434de47171c309ecc2cd06ff98bb39b1237, adding to `api_refinement` seems like the solution.

### Steps to reproduce

```python
import cv2
import numpy as np

img = np.zeros((100, 100), dtype=np.uint8)
cv2.SIFT.create().detectAndCompute(img, None)
```


```sh
$ python3 repro.py
$ mypy repro.py
repro.py:5: error: No overload variant of "detectAndCompute" of "Feature2D" matches argument types "ndarray[tuple[int, int], dtype[unsignedinteger[_8Bit]]]", "None"  [call-overload]
repro.py:5: note: Possible overload variants:
repro.py:5: note:     def detectAndCompute(self, image: Mat | ndarray[Any, dtype[integer[Any] | floating[Any]]], mask: Mat | ndarray[Any, dtype[integer[Any] | floating[Any]]], descriptors: Mat | ndarray[Any, dtype[integer[Any] | floating[Any]]] | None = ..., useProvidedKeypoints: bool = ...) -> tuple[Sequence[KeyPoint], Mat | ndarray[Any, dtype[integer[Any] | floating[Any]]]]
repro.py:5: note:     def detectAndCompute(self, image: UMat, mask: UMat, descriptors: UMat | None = ..., useProvidedKeypoints: bool = ...) -> tuple[Sequence[KeyPoint], UMat]
```

### Issue submission checklist

- [x] I report the issue, it's not a question
- [x] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [x] I updated to the latest OpenCV version and the issue is still there
- [x] There is reproducer code and related data files (videos, images, onnx, etc)

--- a/modules/python/src2/typing_stubs_generation/api_refinement.py
+++ b/modules/python/src2/typing_stubs_generation/api_refinement.py
@@ -340,6 +340,7 @@ NODES_TO_REFINE = {
     SymbolName(("cv", ), (), "resize"): make_optional_arg("dsize"),
     SymbolName(("cv", ), (), "calcHist"): make_optional_arg("mask"),
     SymbolName(("cv", ), (), "floodFill"): make_optional_arg("mask"),
+    SymbolName(("cv", ), ("Feature2D", ), "detectAndCompute"): make_optional_arg("mask"),
     SymbolName(("cv", ), (), "imread"): make_optional_none_return,
     SymbolName(("cv", ), (), "imdecode"): make_optional_none_return,
 }

